src/components/learner-credit-management/requests-tab/data/hooks/useBnrSubsidyRequests.js

```diff
@@ -0,0 +1,176 @@
+import {
+  useCallback, useMemo, useState,
+} from 'react';
+import { logError } from '@edx/frontend-platform/logging';
+import { camelCaseObject } from '@edx/frontend-platform/utils';
+import { debounce } from 'lodash-es';
+import EnterpriseAccessApiService from '../../../../../data/services/EnterpriseAccessApiService';
+import { REQUEST_TAB_VISIBLE_STATES } from '../constants';
+
+// Map table column accessors to API field names for sorting
+const API_FIELDS_BY_TABLE_COLUMN_ACCESSOR = {
+  requestDetails: 'email',
+  amount: 'amount',
+  requestDate: 'requestDate',
+  requestStatus: 'requestStatus',
+};
+
+// Hardcoded filter choices for request status
+// This is a temporary solution until the backend supports dynamic filtering
+const HARDCODED_REQUEST_STATUS_FILTER_CHOICES = [
+  {
+    name: 'Requested',
+    value: 'requested',
+    number: 0,
+  },
+  {
+    name: 'Approved',
+    value: 'approved',
+    number: 0,
+  },
+  {
+    name: 'Declined',
+    value: 'declined',
+    number: 0,
+  },
+];
+
+const applySortByToOptions = (sortBy, options) => {
+  const orderingStrings = sortBy.map(({ id, desc }) => {
+    const apiFieldForColumnAccessor = API_FIELDS_BY_TABLE_COLUMN_ACCESSOR[id];
+    if (!apiFieldForColumnAccessor) {
+      logError(`useBnrSubsidyRequests was unable to find an API field for table column accessor: ${id}`);
+      return undefined;
+    }
+    return desc ? `-${apiFieldForColumnAccessor}` : apiFieldForColumnAccessor;
+  }).filter(Boolean);
+
+  if (orderingStrings.length > 0) {
+    Object.assign(options, {
+      ordering: orderingStrings.join(','),
+    });
+  }
+};
+
+const applyFiltersToOptions = (filters, options) => {
+  const emailSearchQuery = filters?.find(filter => filter.id === 'email')?.value;
+  const statusFilter = filters?.find(filter => filter.id === 'requestStatus')?.value;
+
+  if (emailSearchQuery) {
+    Object.assign(options, {
+      search: emailSearchQuery,
+    });
+  }
+
+  if (statusFilter && statusFilter.length > 0) {
+    Object.assign(options, {
+      state: statusFilter.join(','),
+    });
+  }
+};
+
+// Transform API response data to match DataTable the requirements
+const transformApiDataToTableData = (apiResults) => apiResults.map((item) => {
+  const requestDate = new Date(item.created).toLocaleDateString('en-US', {
+    year: 'numeric',
+    month: 'short',
+    day: 'numeric',
+  });
+
+  return {
+    uuid: item?.uuid,
+    email: item?.email,
+    courseTitle: item?.courseTitle,
+    courseId: item?.courseId,
+    amount: 500,
```

Comment:
[FYI] We will replace this after we add the amount in the `LearnerCreditRequests` Table.